hugolib/resource_chain_test.go

```diff
@@ -424,12 +437,16 @@ CSS integrity Data last:  {{ $cssFingerprinted2.RelPermalink }} {{ $cssFingerpri
 		b.Build(BuildCfg{})
 
 		b.AssertFileContent("public/index.html",
-			`
+			fmt.Sprintf(`
 SUNSET: images/sunset.jpg|/images/sunset.a9bf1d944e19c0f382e0d8f51de690f7d0bc8fa97390c4242a86c3e5c0737e71.jpg|900|90587
 FIT: images/sunset.jpg|/images/sunset_hu59e56ffff1bc1d8d122b1403d34e039f_90587_200x200_fit_q75_box.jpg|200
 CSS integrity Data first: sha256-od9YaHw8nMOL8mUy97Sy8sKwMV3N4hI3aVmZXATxH&#43;8= /styles.min.a1df58687c3c9cc38bf26532f7b4b2f2c2b0315dcde212376959995c04f11fef.css
 CSS integrity Data last:  /styles2.min.1cfc52986836405d37f9998a63fd6dd8608e8c410e5e3db1daaa30f78bc273ba.css sha256-HPxSmGg2QF03&#43;ZmKY/1t2GCOjEEOXj2x2qow94vCc7o=
-`)
+
+SUNSET REMOTE: %[1]s.jpg|/%[1]s.a9bf1d944e19c0f382e0d8f51de690f7d0bc8fa97390c4242a86c3e5c0737e71.jpg|900|90587
+FIT REMOTE: %[1]s.jpg|/%[1]s_hu59e56ffff1bc1d8d122b1403d34e039f_0_200x200_fit_q75_box.jpg|200
```

Comment:
@bep The size in the permalink turns 0, any idea why this could happen? The size is not 0.

I tested fit and resizing and the following works:
```
    {{- $image := resources.FromRemote "https://d33wubrfki0l68.cloudfront.net/291588ca4ec28632c45707669f113eced92ea64d/f1a39/images/homepage-screenshot-hugo-themes.jpg" -}}
    {{ $fit := $image.Fit "200x200"}}

    {{$image.Permalink}}
    {{$fit.Permalink}}

    <img src="{{ $image.Permalink }}" alt="{{ $image.Permalink }}" />
    <img src="{{ $fit.Permalink }}" alt="{{ $fit.Permalink }}" />
```

Reply:
Fixed usign c8b07b4

Reply:
@bep `hugio.NewReadSeekerNoOpCloser` is not storing any files, causing the unavailability of `hugofs.FileMetaInfo` in https://github.com/gohugoio/hugo/blob/master/resources/resource.go#L644, resulting in the 0 source-file size indicator in the publication filename.

I'm not sure why the original file size is included in the filename and if we really care about its or the `hugofs.FileMetaInfo` absence. The `resources.FromString` function has the same limitation (but does not support the image type).

Reply:
The size is 0 because it's not a file (we read that from the FileInfo object we get when doing `os.Stat`). It's probably fixable (we could probably read the size when we generate the hash), but not important.

Reply:
Thanks for your feedback, I will mark this as resolved.